### PR TITLE
Adding exit codes to uploadNeuroDB/bin scripts

### PR DESCRIPTION
**Pull request by cmadjar**
### Added the exit codes to the following scripts in uploadNeuroDB/bin:
- concat_mri.pl
- minc2jiv.pl

### Notes: 
- no exits are present in Mincinfo_wrapper
- did not do mincpik as its future might be compromised once we discuss PR #288. Can always do it later on.

Link to redmine ticket: https://redmine.cbrain.mcgill.ca/issues/14054

### CODE REVIEW

File: uploadNeuroDB/NeuroDB/ExitCodes.pm [hunk 1]
```diff
@@ -105,8 +108,10 @@ our $ARG_FILE_DOES_NOT_EXIST = 5; # if file given as an argument does not exist
 # database related failure
 # called from minc_insertion.pl & register_processed_data.pl
 our $FILE_NOT_UNIQUE = 6; # if file to register is not unique & already
-                           # inserted
+                          # inserted
 
+# other generic failure
+our $FILE_OR_FOLDER_DOES_NOT_EXIST = 6; # if file or folder does not exist
```
Comment (MounaSafiHarab) [inline uploadNeuroDB/NeuroDB/ExitCodes.pm]: just curious at using exit code =6 when 2 lines before we have the same one? is this intentional?

File: uploadNeuroDB/bin/minc2jiv.pl [hunk 2]
```diff
@@ -93,7 +94,8 @@
 my $compress= ($gzip ? "| gzip -c9 " : "") ;
 
 my %base_names_seen;
-MNI::FileUtilities::check_output_path("$TmpDir/") or exit 1;
+MNI::FileUtilities::check_output_path("$TmpDir/")
```
Comment (MounaSafiHarab) [inline uploadNeuroDB/bin/minc2jiv.pl]: wouldn't exit code 5 also work here?

**Comment by MounaSafiHarab** [comment]
@cmadjar 
I put 2 comments earlier
and more importantly, this needs rebasing now!

File: uploadNeuroDB/NeuroDB/ExitCodes.pm (hunk 1 above)
Comment (cmadjar) [reply uploadNeuroDB/NeuroDB/ExitCodes.pm]: Oups, you're right, it should have been 7. I did not use the exit 5 as the exit 5 is specific to files passed as an *argument* that do not exist and here, the file or folder was not provided as an argument. Does that make sense?

File: uploadNeuroDB/bin/minc2jiv.pl (hunk 2 above)
Comment (cmadjar) [reply uploadNeuroDB/bin/minc2jiv.pl]: exit code 5 is specific to files provided as argument to the script which is why I did not use it here.

**Comment by cmadjar** [comment]
@MounaSafiHarab I had to rebase it and change the code number to 9 as there was a conflict with the minor branch. Fixed now. I might have to do the same for the other exit code PR.

**Comment by nicolasbrossard** [comment]
Makes sense to me.

**Comment by MounaSafiHarab** [comment]
> @nicolasbrossard @MounaSafiHarab I propose that we merge this PR if there is no other issue than the fact that exit code should be between 0-255.
> 
> In the meantime, I am going to create a spreadsheet that I will share with you to propose some regroupment of the exit codes to make them more concise. Since the exit code will go to 19.1, it might be good to have the final version in that release. What do you think?

I would for sure say we should aim for 19.1 as the main change is going in there.
I am not sure how much work is involved, but if it doable, for sure we should aim for it (we do not want to tell the users the exit codes changed and then changed again in a next release, if we can).
Just got in, so I will read your proposal!

**Comment by cmadjar** [comment]
> I would for sure say we should aim for 19.1 as the main change is going in there.
I am not sure how much work is involved, but if it doable, for sure we should aim for it (we do not want to tell the users the exit codes changed and then changed again in a next release, if we can).
Just got in, so I will read your proposal!

I totally agree with you. As soon as we agree on the exit codes, I'll make the change and send a PR. I don't think it will be a long one to do. :)

**Comment by MounaSafiHarab** [comment]
> I totally agree with you. As soon as we agree on the exit codes, I'll make the change and send a PR. I don't think it will be a long one to do. :)

sounds great!
I put a couple of comments on the excel sheet; just minor ones to make things even more generic; perhaps @nicolasbrossard can comment there too.
We can also meet briefly in the next day or 2 if you are here if it makes this faster for you!

**Comment by MounaSafiHarab** [comment]
and yes, I agree we cna merge this and make the change in one other complete PR
@nicolasbrossard what do you think? you had requested changes, so we can not move forward until you dismiss/resolve those.

**Comment by nicolasbrossard** [comment]
Ok, I'll approve this PR and we'll tackle the refactoring of the exit codes in another one.

**Comment by MounaSafiHarab** [comment]
ok so merging and will wait for the new PR to hopefully merge it onto this same release.

**Pull request merged**